dagshub/auth/tokens.py

```diff
@@ -27,13 +27,34 @@ def _token_cache(self):
         return self.__token_cache
 
     def add_token(self, token: Dict, host: str = None):
+        """
+        Add a token to the token storage for the specified host.
+
+        Args:
+            token (Dict): A dictionary containing the token information.
+            host (str, optional): The host URL for which the token is being added.
+                If not provided, the configuration's default host URL will be used.
+        """
         host = host or config.host
         if host not in self._token_cache:
             self._token_cache[host] = []
         self._token_cache[host].append(token)
         self._store_cache_file()
 
     def get_token(self, host: str = None, fail_if_no_token: bool = False, **kwargs):
+        """
+        Retrieve a token from the token storage for the specified host.
```

Comment:
Only two public auth functions are `get_token()` and `get_authenticator()`. The difference between them is that `get_token()` returns a plaintext token that can be pasted into MLflow/etc, and the authenticator is a fancier object that has renegotiation logic built in (even though it can't be used with _every_ http client for python)